src/NServiceBus.Core/Transports/OutgoingMessage.cs

```diff
@@ -0,0 +1,32 @@
+namespace NServiceBus.Transports
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// The message going out to the transport
+    /// </summary>
+    public class OutgoingMessage
+    {
+        /// <summary>
+        /// Constructs the message
+        /// </summary>
+        /// <param name="headers">The headers associated with this message</param>
+        /// <param name="body">The body of the message</param>
+        public OutgoingMessage(Dictionary<string, string> headers,byte[] body)
```

Comment:
Agree it was difficult for me to see the benefit of stream in the outgoing
message. But not in general. In the receiver side it can be beneficial. The
question is: Are there any transports which support streaming of outgoing
messages?